src/test/java/liuni/TwitterServiceTest.java

```diff
@@ -0,0 +1,171 @@
+        when(user.getScreenName()).thenReturn(testUserScreenName);
+        when(user.getProfileImageURL()).thenReturn(testUserProfileURL);
+        when(status.getUser()).thenReturn(user);
+    }
+
+    private void setUp_User_GoodURL() {
+        testUserName = "TestUserName";
+        testUserScreenName = "TestUserScreenName";
+        testUserProfileURL = "http://abs.twimg.com/sticky/default_profile_images/default_profile_normal.png";
+
+        testURL = null;
+        try {
+            testURL = new URL(testUserProfileURL);
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+
+        assertTrue(testURL != null);
+
+        status = mock(Status.class);
+        User user = mock(User.class);
+        when(user.getName()).thenReturn(testUserName);
+        when(user.getScreenName()).thenReturn(testUserScreenName);
+        when(user.getProfileImageURL()).thenReturn(testUserProfileURL);
+        when(status.getUser()).thenReturn(user);
+    }
+
+    @Test
+    public void test_getUser_BadURL() {
+        setUp_User_BadURL();
+        UserModel resultModel = twitterService.getUser(status);
+        assertEquals(testUserName, resultModel.getName());
+        assertEquals(testUserScreenName, resultModel.getTwitterHandle());
+        assertEquals(null, resultModel.getProfileImageUrl());
+    }
+
+    @Test
+    public void test_getUser_GoodURL() {
+        setUp_User_GoodURL();
+        UserModel resultModel = twitterService.getUser(status);
+        assertEquals(testUserName, resultModel.getName());
+        assertEquals(testUserScreenName, resultModel.getTwitterHandle());
+        assertEquals(testURL, resultModel.getProfileImageUrl());
+    }
+
+    private void setUp_Tweet() {
+        date = mock(Date.class);
+        testMessage = "Test Message";
+        when(status.getText()).thenReturn(testMessage);
+        when(status.getCreatedAt()).thenReturn(date);
+    }
+
+    @Test
+    public void test_getTweet() {
+        setUp_User_BadURL();
+        setUp_Tweet();
+
+        TwitterTweetModel tweetModel = twitterService.getTweet(status);
+        assertEquals(date, tweetModel.getCreatedAt());
+        assertEquals(testMessage, tweetModel.getMessage());
+        assertEquals(testUserName, tweetModel.getUser().getName());
+        assertEquals(testUserScreenName, tweetModel.getUser().getTwitterHandle());
+        assertEquals(null, tweetModel.getUser().getProfileImageUrl());
+    }
+
+    private TwitterTweetModel createTweetModel() {
+        TwitterTweetModel tweetModel = new TwitterTweetModel();
+        tweetModel.setCreatedAt(date);
+        tweetModel.setMessage(testMessage);
+
+        UserModel user = new UserModel();
+        user.setProfileImageUrl(null);
+        user.setTwitterHandle(testUserScreenName);
+        user.setName(testUserName);
+
+        tweetModel.setUser(user);
+
+        return tweetModel;
+    }
+
+    @Test
+    public void test_convertList() {
```

Comment:
test multiple items in the list and not just for one

Reply:
Done